lib/hqmf-parser/2.0/data_criteria.rb

```
@@ -303,6 +221,20 @@ def extract_type_from_template_id
         if defs
           @definition = defs['definition']
           @status = defs['status'].length > 0 ? defs['status'] : nil
+          # Only attempt to grab settings if no status is present
+          begin
+            settings = HQMF::DataCriteria.get_settings_for_definition(defs['definition'], defs['status'])
+            if settings
+              # apply information from settings to the data criteria
+              @title = settings["title"] if !settings["title"].blank?
+              @definition = settings["definition"] if !settings["definition"].blank?
+              @status = settings["status"] if !settings["status"].blank?
+              @category = settings["category"] if !settings["category"].blank?
+              @hard_status = settings["hard_status"]
+            end
+          rescue Exception
+            # ignore error (none should exist)
```

If we don't expect an error, then why the rescue? In cases like that I'd rather leave out the rescue and, if there is at some point an error, we'll find out about it, which I think is preferable to it being silently ignored.

PS I agree about the end unless @status being confusing, when reading the code linearly it can easily be missed. A good rule of thumb is to only use post-conditionals on fairly short simple lines.